clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/description/AbstractCloudFoundryServerGroupDescription.java

```diff
@@ -16,13 +16,30 @@
 
 package com.netflix.spinnaker.clouddriver.cloudfoundry.deploy.description;
 
+import com.netflix.spinnaker.clouddriver.security.resources.ApplicationNameable;
+import com.netflix.spinnaker.moniker.Moniker;
+import java.util.Collection;
+import java.util.Collections;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 
 @Data
 @EqualsAndHashCode(callSuper = true)
 public abstract class AbstractCloudFoundryServerGroupDescription
-    extends AbstractCloudFoundryDescription {
+    extends AbstractCloudFoundryDescription implements ApplicationNameable {
   private String serverGroupId;
   private String serverGroupName;
+
+  private String cluster;
+  private Moniker moniker;
+
+  @Override
+  public Collection<String> getApplications() {
+    if (moniker != null) {
+      return Collections.singletonList(moniker.getApp());
+    } else if (cluster != null) {
+      return Collections.singletonList(cluster.split("-")[0]);
```

Comment:
Would use Frigga library for this. If for no other reason, it will make it more obvious what's going on here.

Reply:
@danielpeach Thanks for pointing the frigga library out to me! Done